src/main/java/meditracker/command/ListCommand.java

```diff
@@ -66,38 +65,67 @@ public void execute() {
         boolean isAfternoon = parsedArguments.get(ArgumentName.AFTERNOON) != null;
         boolean isEvening = parsedArguments.get(ArgumentName.EVENING) != null;
         Period period = Period.getPeriod(isMorning, isAfternoon, isEvening);
-        
-        switch (listTypeString) {
-        case "all":
-            MedicationManager.printAllMedications();
-            break;
-        case "today":
-            switch (period) {
-            case MORNING:
-                List<DailyMedication> morningMedications = DailyMedicationManager.getDailyMedications(Period.MORNING);
-                DailyMedicationManager.printTodayMedications(MedicationManager.getMedications(),
-                        morningMedications, "Morning:");
-                break;
-            case AFTERNOON:
-                List<DailyMedication> afternoonMedications
-                        = DailyMedicationManager.getDailyMedications(Period.AFTERNOON);
-                DailyMedicationManager.printTodayMedications(MedicationManager.getMedications(),
-                        afternoonMedications, "Afternoon:");
-                break;
-            case EVENING:
-                List<DailyMedication> eveningMedications = DailyMedicationManager.getDailyMedications(Period.EVENING);
-                DailyMedicationManager.printTodayMedications(MedicationManager.getMedications(),
-                        eveningMedications, "Evening:");
+
+        // checks if user added extra flags or words after "list -t all"
+        if (listTypeString.equals("all") && parsedArguments.size() > 1) {
+            Ui.showErrorMessage(String.format("List type -> \"%s\" not compatible with " +
+                    "\"list -t all\" command.", period));
+        } else {
+            switch (listTypeString) {
+            case "all":
```

Comment:
Might be better to move L70 condition check into this switch case so the switch statement does not have to be nested an additional layer

```suggestion
            case "all":
                if (parsedArguments.size() > 1) {
                    Ui.showErrorMessage(String.format("List type -> \"%s\" not compatible with " +
                    "\"list -t all\" command.", period));
                    return;
                }
```

Reply:
Thank you for the suggestion, implemented in 98ba095